api-scanner/src/main/java/net/corda/plugins/ScanApi.java

```diff
@@ -348,14 +349,10 @@ private Names toNames(Collection<ClassInfo> classes) {
             Map<Boolean, List<String>> partitioned = classes.stream()
                 .map(ClassInfo::toString)
                 .filter(ScanApi::isApplicationClass)
-                .collect(partitioningBy(this::isVisibleAnnotation));
-            /*
-             * These annotations are returned in an order determined by
-             * fast-classpath-scanner. THIS IS BAD! The ordering here
-             * will almost certainly change/break in future versions.
-             * TODO: Order these annotations ourselves!
-             */
-            return new Names(partitioned.get(true), partitioned.get(false));
+                .collect(partitioningBy(this::isVisibleAnnotation, toCollection(LinkedList::new)));
+
+            Function<List<String>, List<String>> ordering = list -> { sort(list); return list; };
```

Comment:
What you had before (method which sorted and then returned the list) was better

Reply:
I thought a "local function" approach would be better precisely because `sort()` returns `void` in Java. Adding a private `sort()` method which returned the list started to have confusing implications for the sorts in the rest of the code. Semantically, wouldn't you expect the new function to be "generally applicable" and so return a sorted _copy_ of the list, rather than the sorted original list?

Basically I dodged the issue by creating a lambda for this very _specific_ use-case only.

Reply:
The `Function` declaration just makes this a bit too verbose, that's all.